src/AppInstallerRepositoryCore/Microsoft/Schema/1_4/DependenciesTable.cpp

```diff
@@ -0,0 +1,375 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "DependenciesTable.h"
+#include "SQLiteStatementBuilder.h"
+#include "winget\DependenciesGraph.h"
+#include "Microsoft/Schema/1_0/OneToOneTable.h"
+#include "Microsoft/Schema/1_0/IdTable.h"
+#include "Microsoft/Schema/1_0/ManifestTable.h"
+#include "Microsoft/Schema/1_0/VersionTable.h"
+#include "Microsoft/Schema/1_0/Interface.h"
+#include "Microsoft/Schema/1_0/ChannelTable.h"
+
+namespace AppInstaller::Repository::Microsoft::Schema::V1_4
+{
+    using namespace AppInstaller;
+    using namespace std::string_view_literals;
+    using namespace SQLite::Builder;
+    using namespace Schema::V1_0;
+    using QCol = SQLite::Builder::QualifiedColumn;
+
+    static constexpr std::string_view s_DependenciesTable_Table_Name = "dependencies"sv;
+    static constexpr std::string_view s_DependenciesTable_Index_Name = "dependencies_index"sv;
+    static constexpr std::string_view s_DependenciesTable_Manifest_Column_Name = "manifest"sv;
+    static constexpr std::string_view s_DependenciesTable_MinVersion_Column_Name = "min_version"sv;
+    static constexpr std::string_view s_DependenciesTable_PackageId_Column_Name = "package_id";
+
+    namespace
+    {
+        std::set<Manifest::Dependency> GetDependencies(
+            const Manifest::Manifest& manifest, Manifest::DependencyType dependencyType)
+        {
+            std::set<Manifest::Dependency>  dependencies;
+
+            for (const auto& installer : manifest.Installers)
+            {
+                installer.Dependencies.ApplyToAll([&](Manifest::Dependency dependency)
+                    {
+                        if (dependency.Type == dependencyType)
+                        {
+                            dependencies.emplace(dependency);
+                        }
+                    });
+            }
+
+            return dependencies;
+        }
+
+        void RemoveDependenciesByRowIds(SQLite::Connection& connection, std::vector<SQLite::rowid_t> dependencyRowIds)
+        {
+            using namespace SQLite::Builder;
+            if (dependencyRowIds.empty())
+            {
+                return;
+            }
+            SQLite::Savepoint savepoint = SQLite::Savepoint::Create(connection, std::string{ s_DependenciesTable_Table_Name } + "remove_dependencies_by_rowid");
+
+            SQLite::Builder::StatementBuilder builder;
+            builder.DeleteFrom(s_DependenciesTable_Table_Name).Where(SQLite::RowIDName).In(dependencyRowIds.size());
```

Comment:
>builder.DeleteFrom(s_DependenciesTable_Table_Name).Where(SQLite::RowIDName).In(dependencyRowIds.size()); [](http://example.com/codeflow?start=12&length=104)

I'm not sure that you are optimizing this like you think you are.  No need to change it now, but I think you did extra work over just creating a statement and executing it N times that probably doesn't help.

Even if it is faster it feels like premature optimization.  No blame here, the struggle is real, but we still need to fight against it.